portfolio/src/main/webapp/script.js

```diff
@@ -12,23 +12,50 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// A list of activities that could appear when clicking activities button.
-const activities = ['Coding 💻', 'Travelling 🗼', 'Brewing coffee ☕', 'Going on a nature walk 🌲'];
+// This is a list of passions to iterate through on the homepage
+const passions = ["Front-End Development?", "Machine Learning?", "Software Design?", "Technology Consulting?" ,"Data Science?", "Algorithms?", "User Experience?"];
+const typewriterLetterDelay = 100;
+const typewriterWordDelay = 1000;
 
+// Get the selector for the Passions type-writer function when the window loads.
+window.onload = function () { 
+    var passionSelector = document.getElementById("passions").childNodes[0]
 
-/**
- * Adds a random activity I might be doing to the page
- */
-function addActivity() {
-  const activityContainer = document.getElementById('activity-container');
+    window.setTimeout(() => typewriter(passionSelector, passions), 1000);
+};
 
-  // Always show a new activity from the below list of activities
-  // every time the button is clicked.
-  const eligibleActivities = activities.filter(ele => ele !== activityContainer.innerText);
+// Function to start a typewriter effect on an HTML element
+// where one letter of a word (in a list of words) appears at a time
+// @param textSelector - html selector of target element
+// @param words - list of words to cycle through in the effect. min length is 1
+// @param letterDelay - duration in milliseconds of delay between letter keypresses. Must be > 0
+// @param wordDelay - duration in milliseconds of delay between presenting a finished word and starting the next one. Must be >0
```

Comment:
Thanks - Actually removed the default values. After more thought it makes the function less reusable, in case if I want to use different typewriter effects later with different delays. No reason for the first implementation to be default and then the later ones custom.